src/GitVersionExe/Program.cs

```diff
@@ -129,17 +129,27 @@ static void ConfigureLogging(Arguments arguments)
             {
                 try
                 {
-                    Directory.CreateDirectory(Path.GetDirectoryName(arguments.LogFilePath));
-                    if (File.Exists(arguments.LogFilePath))
+                    var logFileFullPath = Path.GetFullPath(arguments.LogFilePath);
+                    var logFile = new FileInfo(logFileFullPath);
+
+                    if (logFile.Directory == null)
```

Comment:
Thanks for pointing this out. After some investigation, it seems like for paths like `C:\GitVersion.log`, `FileInfo.Directory` will be `null`. In such circumstances, the file should still be possible to create and it is indeed not a failure scenario to place the file on root. I'll rework the code.